lib/omit-properties/index.ts

```diff
@@ -1,3 +1 @@
-import { PickKeysByValue } from "../pick-keys-by-value";
-
-export type OmitProperties<Type, Value> = Omit<Type, PickKeysByValue<Type, Value>>;
+export type OmitProperties<Type, Value> = { [Key in keyof Type as Type[Key] extends Value ? never : Key]: Type[Key] };
```

Comment:
Amazing work! It's very performant right now

<img width="1114" alt="Screenshot 2024-05-19 at 12 20 49" src="https://github.com/ts-essentials/ts-essentials/assets/2991847/02971ec2-5aa3-41dc-80e9-59376da78b0f">